include/deal.II/base/array_view.h

```diff
@@ -286,6 +349,52 @@ make_array_view (std::vector<ElementType> &vector,
 }
 
 
+/**
+ * Create an ArrayView that takes a pair of iterators as arguments. The type
+ * of the ArrayView is inferred from the value type of the iterator (e.g., the
+ * view created from two const iterators will have a const type).
+ *
+ * @warning This function only makes sense for array or vector iterators
+ * (i.e., the memory between @p begin and @end contains valid objects). This
+ * function is intended for use with the various boost vector classes and
+ * <code>std::vector</code>, but not, e.g., <code>std::deque</code>.
+ *
+ * @relates ArrayView
+ */
+template <typename Iterator>
+ArrayView<typename std::remove_reference<typename std::iterator_traits<Iterator>::reference>::type>
+make_array_view (const Iterator begin, const Iterator end)
```

Comment:
if you think of cases such as `tria.begin_active()` and `tria.end()`, then the two types don't actually have to be the same. The latter just has to be convertible to the former. We typically do this by making the second argument of type `identity<Iterator>::type`.